python/sglang/srt/layers/quantization/modelopt_quant.py

```diff
@@ -642,10 +642,21 @@ def from_config(cls, config: Dict[str, Any]) -> ModelOptFp4Config:
     def is_layer_excluded(self, prefix: str, exclude_modules: list):
         import regex as re
 
+        fused_patterns = ["q_a_proj", "q_b_proj", "kv_a_proj_with_mqa", "kv_b_proj"]
+        prefix_split = prefix.split(".")
         for pattern in exclude_modules:
             regex_str = pattern.replace(".", r"\.").replace("*", r".*")
+            pattern_split = pattern.split(".")
             if re.fullmatch(regex_str, prefix):
                 return True
+            # Check if the last part of the excluded pattern is contained in the last part of the prefix
+            # This handles fused modules like fused_qkv_a_proj_with_mqa that contain q_a_proj and kv_a_proj_with_mqa
+            # e.g., model.layers.{i}.self_attn.{fused_weight_name}
+            elif len(prefix_split) == 5 and (
```

Comment:
would be great to avoid "5" etc and make code a bit more spcific, e.g. maybe assert the pattern

Reply:
agree